src/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import styled from '@emotion/styled';
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { useCallback } from 'react';
+import { Spinner } from '../spinner';
+import { CheckboxIcon } from './CheckboxIcon';
+
+interface Props {
+  checked: boolean | 'indeterminate';
+  className?: string;
+
+  label?: string;
+  ariaLabel?: string;
+  title?: string;
+  id?: string;
+
+  isDisabled?: boolean;
+  isLoading?: boolean;
+
+  onCheck: (checked: boolean | 'indeterminate', id?: string) => void;
+  onFocus?: VoidFunction;
+}
+
+export function Checkbox(props: Readonly<Props>) {
+  const { checked, className, label, ariaLabel, title, id, isDisabled, isLoading = false } = props;
+  const { onCheck, onFocus } = props;
+
+  const handleChange = useCallback(
+    (checked: boolean | 'indeterminate') => {
+      if (!isDisabled && !isLoading) {
+        onCheck(checked, id);
+      }
+    },
+    [isDisabled, isLoading, onCheck, id],
+  );
+
+  return (
+    <CheckboxContainer
+      aria-disabled={isDisabled}
+      as={label ? 'label' : 'span'}
+      className={className}>
+      <Spinner isLoading={isLoading}>
+        <CheckboxRoot
+          aria-disabled={isDisabled}
+          aria-label={ariaLabel ?? title}
+          checked={checked}
+          id={id}
+          onCheckedChange={handleChange}
+          onFocus={onFocus}
+          title={title}>
+          <CheckboxIndicator>
+            <CheckboxIcon checked={checked} />
+          </CheckboxIndicator>
+        </CheckboxRoot>
+      </Spinner>
+      {label && <Label>{label}</Label>}
+    </CheckboxContainer>
+  );
+}
+
+const CheckboxContainer = styled.span`
```

Comment:
NIT: In SQ we are also using "user-select: none", I feel its not needed but just putting it here

Reply:
Yeah same in SC and I'm not sure it really makes sense to prevent selecting the label next to the checkbox. I would keep it like this.